metals/src/main/scala/scala/meta/internal/metals/debug/SocketEndpoint.scala

```diff
@@ -3,22 +3,34 @@ package scala.meta.internal.metals.debug
 import java.io.BufferedInputStream
 import java.io.BufferedOutputStream
 import java.net.Socket
-import java.util.Collections
 
-import scala.meta.internal.metals.debug.SocketEndpoint._
+import scala.collection.concurrent.TrieMap
 
+import org.eclipse.lsp4j.debug.services.IDebugProtocolServer
 import org.eclipse.lsp4j.jsonrpc.MessageConsumer
 import org.eclipse.lsp4j.jsonrpc.debug.json.DebugMessageJsonHandler
+import org.eclipse.lsp4j.jsonrpc.json.MethodProvider
 import org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer
 import org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer
 import org.eclipse.lsp4j.jsonrpc.messages.Message
+import org.eclipse.lsp4j.jsonrpc.messages.RequestMessage
+import org.eclipse.lsp4j.jsonrpc.services.ServiceEndpoints
 
 private[debug] final class SocketEndpoint(socket: Socket)
     extends RemoteEndpoint {
+  private val methods =
```

Comment:
This is how json rpc now knows how to deserialize message body. Without it the new version stopped returning anything. `getSupportedMethods` just gets all the annotated methods from a specified interface.